Mathlib/RingTheory/Valuation/Basic.lean

```diff
@@ -0,0 +1,926 @@
+  intro x y
+  by_cases hy : y = 0; · simp [hy, zero_iff]
+  rw [show y = 1 * y by rw [one_mul]]
+  rw [← inv_mul_cancel_right₀ hy x]
+  iterate 2 rw [v.map_mul _ y, v'.map_mul _ y]
+  rw [v.map_one, v'.map_one]
+  constructor <;> intro H
+  · apply mul_le_mul_right'
+    replace hy := v.ne_zero_iff.mpr hy
+    replace H := le_of_le_mul_right hy H
+    rwa [h] at H
+  · apply mul_le_mul_right'
+    replace hy := v'.ne_zero_iff.mpr hy
+    replace H := le_of_le_mul_right hy H
+    rwa [h]
+#align valuation.is_equiv_of_val_le_one Valuation.IsEquiv_of_val_le_one
+
+theorem IsEquiv_iff_val_le_one [LinearOrderedCommGroupWithZero Γ₀]
+    [LinearOrderedCommGroupWithZero Γ'₀] (v : Valuation K Γ₀) (v' : Valuation K Γ'₀) :
+    v.IsEquiv v' ↔ ∀ {x : K}, v x ≤ 1 ↔ v' x ≤ 1 :=
+  ⟨fun h x => by simpa using h x 1, IsEquiv_of_val_le_one _ _⟩
+#align valuation.is_equiv_iff_val_le_one Valuation.IsEquiv_iff_val_le_one
+
+theorem IsEquiv_iff_val_eq_one [LinearOrderedCommGroupWithZero Γ₀]
+    [LinearOrderedCommGroupWithZero Γ'₀] (v : Valuation K Γ₀) (v' : Valuation K Γ'₀) :
+    v.IsEquiv v' ↔ ∀ {x : K}, v x = 1 ↔ v' x = 1 := by
+  constructor
+  · intro h x
+    simpa using @IsEquiv.val_eq _ _ _ _ _ _ v v' h x 1
+  · intro h
+    apply IsEquiv_of_val_le_one
+    intro x
+    constructor
+    · intro hx
+      cases' lt_or_eq_of_le hx with hx' hx'
+      · have : v (1 + x) = 1 := by
+          rw [← v.map_one]
+          apply map_add_eq_of_lt_left
+          simpa
+        rw [h] at this
+        rw [show x = -1 + (1 + x) by simp]
+        refine' le_trans (v'.map_add _ _) _
+        simp [this]
+      · rw [h] at hx'
+        exact le_of_eq hx'
+    · intro hx
+      cases' lt_or_eq_of_le hx with hx' hx'
+      · have : v' (1 + x) = 1 := by
+          rw [← v'.map_one]
+          apply map_add_eq_of_lt_left
+          simpa
+        rw [← h] at this
+        rw [show x = -1 + (1 + x) by simp]
+        refine' le_trans (v.map_add _ _) _
+        simp [this]
+      · rw [← h] at hx'
+        exact le_of_eq hx'
+#align valuation.is_equiv_iff_val_eq_one Valuation.IsEquiv_iff_val_eq_one
+
+theorem IsEquiv_iff_val_lt_one [LinearOrderedCommGroupWithZero Γ₀]
+    [LinearOrderedCommGroupWithZero Γ'₀] (v : Valuation K Γ₀) (v' : Valuation K Γ'₀) :
+    v.IsEquiv v' ↔ ∀ {x : K}, v x < 1 ↔ v' x < 1 := by
+  constructor
+  · intro h x
+    simp only [lt_iff_le_and_ne,
+      and_congr ((IsEquiv_iff_val_le_one _ _).1 h) ((IsEquiv_iff_val_eq_one _ _).1 h).not]
+  · rw [IsEquiv_iff_val_eq_one]
+    intro h x
+    by_cases hx : x = 0
+    · -- porting note: this proof was `simp only [(zero_iff _).2 hx, zero_ne_one]`
+      rw [(zero_iff _).2 hx, (zero_iff _).2 hx]
+      simp only [zero_ne_one]
+    constructor
+    · intro hh
+      by_contra h_1
+      cases ne_iff_lt_or_gt.1 h_1 with
+      | inl h_2 => simpa [hh, lt_self_iff_false] using h.2 h_2
+      | inr h_2 =>
+          rw [← inv_one, eq_inv_iff_eq_inv, ← map_inv₀] at hh
```

Comment:
`eq_inv_iff_eq_inv` was removed by #17483. Use `<- inv_eq_iff_eq_inv` instead. (Same in line 528.)